src/textual/css/model.py

```diff
@@ -94,6 +94,13 @@ def _check_universal(self, node: DOMNode) -> bool:
     def _check_type(self, node: DOMNode) -> bool:
```

Comment:
do we need to add this to `_check_id` or `_check_class` also? 

For `_check_id` we could have a match on the id, then if the element is found, there could be an extra check whether it is the first child (or the last child, depending on the selector), but I'm not sure if it would be the correct behavior.

For `_check_class` I'm thinking on the usage and the test and the following didn't work:
`app.query(Child).filter(":first-child")`

Reply:
I'm not sure how the semantics for the pseudo selectors work.
Maybe @willmcgugan can clarify.

Reply:
`app.query("Child:first_child", Child)` should work.

Reply:
Maybe I miss something but `app.query("Child:first_child", Child)` does not work for me.

Is this the correct function to be called?

Reply:
> Maybe I miss something but `app.query("Child:first_child", Child)` does not work for me.

I think what Will is trying to say is that when `:first_child` is implemented properly, the query above should be made to work.
If you've implemented the pseudo-selector `:first_child` and the call above doesn't work, then something needs to be fixed.